saleor/graphql/channel/mutations/channel_update.py

```diff
@@ -84,9 +90,27 @@ def clean_input(cls, info: ResolveInfo, instance, data, **kwargs):
             cleaned_input["slug"] = slugify(slug)
         if stock_settings := cleaned_input.get("stock_settings"):
             cleaned_input["allocation_strategy"] = stock_settings["allocation_strategy"]
+        if order_settings := cleaned_input.get("order_settings"):
+            cleaned_input["automatically_confirm_all_new_orders"] = order_settings[
+                "automatically_confirm_all_new_orders"
+            ]
+            cleaned_input[
+                "automatically_fulfill_non_shippable_gift_card"
+            ] = order_settings["automatically_fulfill_non_shippable_gift_card"]
 
         return cleaned_input
 
+    @classmethod
+    def check_permissions(cls, context, permissions=None, **data):
+        permissions = [ChannelPermissions.MANAGE_CHANNELS]
+
+        # when updating only orderSettings allow MANAGE_ORDERS permission
```

Comment:
This comment is a little bit incomprehensible. Could you try to rewrite it?

Reply:
Sure thing. What do you think about:
`Permission MANAGE_ORDERS or MANAGE_CHANNELS is required to update orderSettings.
MANAGE_ORDERS can be used only when channelUpdate mutation will get OrderSettingsInput and not any other fields. Otherwise needed permission is MANAGE_CHANNELS`

Reply:
Sounds good to me 👍